shamil_latypov/logger.cpp

```diff
@@ -0,0 +1,26 @@
+#include "logger.hpp"
+#include <assert.h>
+#include <cassert>
+#include <chrono>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+namespace uni_cource_cpp {
+
+void Logger::set_file_output_name(const std::string& file_path) {
+  file_out_ = std::ofstream(file_path);
+  assert(file_out_.is_open() && "File didnt open. Maybe Temp dir soesnt exist");
+}
+
+void Logger::log(const std::string& text) {
+  std::cout << text << std::endl;
+  file_out_ << text << std::endl;
```

Comment:
И куда-же он пишет по-дефолту?
Нужно проверить, что стрим существует.
```suggestion
  if (file_out_.has_value()) {
    file_out_.value() << text << std::endl;
  }
```